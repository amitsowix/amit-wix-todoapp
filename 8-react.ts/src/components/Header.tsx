import {FC} from 'react';
import {createUseStyles} from 'react-jss'
import {Classes} from 'jss';

const Header: FC = () => {
    const classes = useStyles();
    return (
        <div className = {classes.appHeader}>
            {'Welcome to the most AMAZING\n(written in React.ts) Todo App EVERRRRR!'}
        </div>
    )            
}

const useStyles: () => Classes = createUseStyles({
    appHeader: {
        height: "30%",
        background: "indianred",
        borderBottomLeftRadius: "50%",
        borderBottomRightRadius: "50%",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        color: "white",
        fontSize: "47px",
        textAlign: "center",
        lineHeight: "70px",
        whiteSpace: 'break-spaces'
    }
});

export default Header;