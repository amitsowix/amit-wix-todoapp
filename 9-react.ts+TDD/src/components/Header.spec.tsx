import React from 'react';
import renderer, {ReactTestRenderer} from 'react-test-renderer';
import Header from './Header';

const driver = {
    when: {
        mount: (): ReactTestRenderer => {
            const component: ReactTestRenderer = renderer.create(<Header/>)
            return component;
        }
    }
}

describe('Header', () => {
    it('should render component', () => {

        const tree: ReactTestRenderer = driver.when.mount();

        expect(tree).toMatchSnapshot();
    })
});