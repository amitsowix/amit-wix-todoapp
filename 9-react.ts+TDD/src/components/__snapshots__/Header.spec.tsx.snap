// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render component 1`] = `
<div
  className="appHeader-0-2-1"
>
  Welcome to the most AMAZING (written in React.ts) Todo App EVERRRRR!
</div>
`;
